Parsedown.php

```diff
@@ -1485,10 +1503,80 @@ function parse($text)
         return $markup;
     }
 
+    protected function sanitiseElement(array $Element)
+    {
+        static $badAttributeChars = "\"'= \t\n\r\0\x0B";
+        static $safeUrlNameToAtt  = array(
+            'a'   => 'href',
+            'img' => 'src',
+        );
+
+        if (isset($safeUrlNameToAtt[$Element['name']]))
+        {
+            $Element = $this->filterUnsafeUrlInAttribute($Element, $safeUrlNameToAtt[$Element['name']]);
+        }
+
+        if ( ! empty($Element['attributes']))
+        {
+            foreach ($Element['attributes'] as $att => $val)
+            {
+                # clear out nulls
+                if ($val === null)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+                # filter out badly parsed attribute
+                elseif (strpbrk($att, $badAttributeChars) !== false)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+            }
+
+            $onEventAttributeKeys = preg_grep('/^on/i', array_keys($Element['attributes']));
+
+            foreach ($onEventAttributeKeys as $att)
+            {
+                unset($Element['attributes'][$att]);
+            }
+        }
+
+        return $Element;
+    }
+
+    protected function filterUnsafeUrlInAttribute(array $Element, $attribute)
+    {
+        if ($this->safeLinksEnabled)
+        {
+            $safe = false;
+
+            foreach ($this->safeLinksWhitelist as $scheme)
+            {
+                if (stripos($Element['attributes'][$attribute], $scheme) === 0)
```

Comment:
\**Striking through what I've written before as you're checking Symfony right now\**

I don't think that we're talking about the same thing, especially regarding whitelists and blacklists. You're talking about sanitizing and validating input (something that is crucial!), not whitelisting. I'm in contrast talking about whitelisting particular protocols like `http://` and `ftp://`, as this is, by definition, something user's will stumble upon (because there's nothing bad about `irc://`, `steam://` and so on - the list is endless, that's problem of trying to whitelist them)

However, it doesn't matter. I think that you'll get enough input by looking at how Symfony solves this problem ([as said](https://github.com/erusev/parsedown/pull/495#discussion_r114670410), encoding is probably the right solution in general, but there are still some open questions) - so I'll simply wait for your next update :smiley: